test/server/listener_manager_impl_test.cc

```diff
@@ -330,6 +330,48 @@ TEST_F(ListenerManagerImplTest, AddListenerAddressNotMatching) {
   EXPECT_CALL(*listener_foo, onDestroy());
 }
 
+TEST_F(ListenerManagerImplTest, UpdateListenerTransparentNotMatching) {
+  InSequence s;
+
+  // Add foo listener.
+  const std::string listener_foo_json = R"EOF(
+  {
+    "name": "foo",
+    "address": "tcp://0.0.0.0:1234",
+    "filters": [],
+    "transparent": true
+  }
+  )EOF";
+
+  ListenerHandle* listener_foo = expectListenerCreate(false);
+  EXPECT_CALL(listener_factory_, createListenSocket(_, true, true));
+  EXPECT_TRUE(manager_->addOrUpdateListener(parseListenerFromJson(listener_foo_json), true));
+  checkStats(1, 0, 0, 0, 1, 0);
+
+  // Update foo listener, but with a different transparent option. Should throw.
```

Comment:
Is there a reason why we cannot disable `IP_TRANSPARENT` for the new listener?

Reply:
@PiotrSikora I was thinking that we have an atomicity issue. How do we atomically hand off the socket to the new listener and set/unset the option? That would require that there is a period of time when no thread is accepting on the socket during the handoff. I don't think we have that (?). And that's not desirable, since that would cause connections to be rejected during that period.

Reply:
@rlenglet there is always a period of time when no thread is accepting new connections (i.e. when all threads are busy processing active requests), and no connections are rejected because of kernel-maintained backlog queue on the listening socket, so I don't think atomicity is the problem.

Reply:
I'm removing that verification. We know from an accepted socket's option whether it has been TPROXYed or not, so we can safely have a mix of connections with and without TPROXY in the same queue.

Reply:
@PiotrSikora There's a bigger challenge with modifying a listener's pre-bind options: when we modify a listener, its listen socket has already been bound, so its pre-bind options can't be modified. If LDS requires those options to be modified, we need to close and re-open the listen socket, and set the new set of options before bind. To avoid losing traffic, that would also require setting `SO_REUSEPORT` on the new socket while the old listener drains.
Would you be open to that?